pilot/pkg/xds/endpoint_builder.go

```diff
@@ -68,26 +71,33 @@ type EndpointBuilder struct {
 	hostname   host.Name
 	port       int
 	push       *model.PushContext
+
+	mtlsChecker *mtlsChecker
 }
 
 func NewEndpointBuilder(clusterName string, proxy *model.Proxy, push *model.PushContext) EndpointBuilder {
 	_, subsetName, hostname, port := model.ParseSubsetKey(clusterName)
 	svc := push.ServiceForHostname(proxy, hostname)
-	return EndpointBuilder{
+	dr := push.DestinationRule(proxy, svc)
+	b := EndpointBuilder{
 		clusterName:     clusterName,
 		network:         proxy.Metadata.Network,
 		networkView:     model.GetNetworkView(proxy),
 		clusterID:       proxy.Metadata.ClusterID,
 		locality:        proxy.Locality,
 		service:         svc,
-		destinationRule: push.DestinationRule(proxy, svc),
+		destinationRule: dr,
 		tunnelType:      GetTunnelBuilderType(clusterName, proxy, push),
 
 		push:       push,
 		subsetName: subsetName,
 		hostname:   hostname,
 		port:       port,
 	}
+	if b.MultiNetworkConfigured() {
+		b.mtlsChecker = newMtlsChecker(push, port, dr)
```

Comment:
How often and when are `NewEndpointBuilder` invoked? in case of new Config change, or endpoints change or both? Relevant for performance and accuracy.

Reply:
Pretty much every config change + endpoints change. Is there a concern with initializing the `mtlsChecker`? The performance impact shouldn't be felt until we actually utilize it.